### PR TITLE
Major reorganisation of the Pop3Store to match the other stores

### PR DESCRIPTION
**Pull request by philipwhiuk**
Highlights:

* Split into many separate classes - Pop3Store now 1000 lines smaller.
* New MockPop3Server similar to MockImapServer
* 73% package test coverage - main un-tested stuff is in the Pop3Folder sync code which I just haven't gotten round to.

### CODE REVIEW

File: k9mail-library/src/test/java/com/fsck/k9/mail/store/pop3/Pop3ConnectionTest.java [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+    private SimplePop3Settings settings;
+    private TrustedSocketFactory socketFactory;
+
+    @Before
+    public void before() throws Exception {
+        settings = new SimplePop3Settings();
+        settings.setUsername(username);
+        settings.setPassword(password);
+        socketFactory = TestTrustedSocketFactory.newInstance();
+        mockTrustedSocketFactory = mock(TrustedSocketFactory.class); //TODO: Remove
```
Comment (cketti) [inline k9mail-library/src/test/java/com/fsck/k9/mail/store/pop3/Pop3ConnectionTest.java]: What's with the TODOs?

File: k9mail-library/src/test/java/com/fsck/k9/mail/store/pop3/Pop3FolderTest.java [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        mockListener = mock(MessageRetrievalListener.class);
+        when(mockStore.getConfig()).thenReturn(mockStoreConfig);
+        when(mockStoreConfig.getInboxFolderName()).thenReturn("Inbox");
+        when(mockStore.createConnection()).thenReturn(mockConnection);
+        when(mockConnection.executeSimpleCommand(Pop3Commands.STAT_COMMAND)).thenReturn("+OK 10 0");
+        folder = new Pop3Folder(mockStore, "Inbox");
+        setupTempDirectory();
+    }
+
+    private void setupTempDirectory() {
```
Comment (cketti) [inline k9mail-library/src/test/java/com/fsck/k9/mail/store/pop3/Pop3FolderTest.java]: In another test class we use `BinaryTempFileBody.setTempDirectory(new File(System.getProperty("java.io.tmpdir")));`. That seems to work well enough.

**Comment by philipwhiuk** [comment]
Thanks. It's been a while since I actually wrote this! I'll tidy up.

**Comment by philipwhiuk** [comment]
@cketti this should be good now :)

**Comment by philipwhiuk** [comment]
Fixes #2731

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/pop3/Pop3Message.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public void setFlag(Flag flag, boolean set) throws MessagingException {
+        super.setFlag(flag, set);
+        mFolder.setFlags(Collections.singletonList(this), Collections.singleton(flag), set);
+    }
+
+    @Override
+    public void delete(String trashFolderName) throws MessagingException {
+        //  try
```
Comment (Valodim) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/pop3/Pop3Message.java]: what's the deal with this try/catch?

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/pop3/Pop3ResponseInputStream.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package com.fsck.k9.mail.store.pop3;
+
+
+import java.io.IOException;
+import java.io.InputStream;
+
+
+class Pop3ResponseInputStream extends InputStream {
+    private InputStream mIn;
+    private boolean mStartOfLine = true;
```
Comment (Valodim) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/pop3/Pop3ResponseInputStream.java]: might as well reformat these to get rid of the m while we touch the code

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/pop3/Pop3Store.java [hunk 5]
```diff
@@ -200,23 +158,15 @@ public static String createUri(ServerSettings server) {
...
-    private boolean mTopNotSupported;
+    private final String host;
+    private final int port;
+    private final String username;
+    private final String password;
+    private final String clientCertificateAlias;
+    private final AuthType authType;
+    private final ConnectionSecurity connectionSecurity;
 
+    private Map<String, Pop3Folder> mFolders = new HashMap<String, Pop3Folder>();
```
Comment (Valodim) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/pop3/Pop3Store.java]: m*

**Pull request merged**